### ISSUE TITLE
Consider including ES5 build

### ISSUE THREAD

**Issue by dbx834** [box/box-ui-elements#184]
Please consider including ES5 build also with `box-ui-elements` and `box-react-ui`.

Box doesn't integrate out of the box(!) at the moment with https://www.meteor.com/. Right now Meteor does not transpile anything in node_modules, so it takes an extra step to get Box working.

See: https://github.com/meteor/meteor-feature-requests/issues/6

**Comment by nateabele** [comment]
Yes, please. Just ran into this today and it's breaking builds. Distributing ES6 code is considered bad practice. Consumers should not have to transpile your code.

**Comment by priyajeet** [comment]
When you guys say ES5, are you still importing via npm/yarn?
In other words, you just want another output in the npm package that does not have import statements?

**Comment by nateabele** [comment]
Yup. `yarn add box-ui-elements`, then:

```javascript
import { ContentExplorer } from 'box-ui-elements';
```

> In other words, you just want another output in the npm package that does not have import statements?

Yup. So in your `package.json`, you have this:

```json
  "main": "lib/index.js",
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
```

The `module` and `jsnext` configs are fine, but only for tools that support that in their default configuration. The `main` entry needs to point to a properly transpiled version of `lib` (i.e. `lib-es5` or whatever). This will ensure your package is compatible out of the box with most peoples' setups, and they won't be required to do any extra configuration to support it.

**Comment by nateabele** [comment]
Also, I realize I misspoke (mistyped?) before: distributing ES6 code is _fine_, so long as it's not your `main`. That's what `module` and `jsnext:main` are for.

**Comment by priyajeet** [comment]
Btw, you ideally should be using full path whenever possible. Tree shaking is not perfect (at least with webpack) and you will get a larger bundle size if you don't.

`import ContentExplorer from 'box-ui-elements/lib/components/ContentExplorer';`

I will likely dump es5 stuff in es5 folder.

`import ContentExplorer from 'box-ui-elements/es5/components/ContentExplorer';`

**Comment by nateabele** [comment]
> Btw, you ideally should be using full path whenever possible.

Huh, that's... inconsistent with my experience. Especially in TypeScript-land, there are a number of popular coding standards that explicitly forbid sub-path imports.

> Tree shaking is not perfect (at least with webpack) and you will get a larger bundle size if you don't.

Don't think I've ever had a problem with this. We keep regular tabs on it with bundle analysis.

**Comment by dbx834** [comment]
_When you guys say ES5, are you still importing via npm/yarn?_

Yes, my group uses NPM mostly.

As I perceive it, the package is a bit opinionated about the tooling and setup people will or are supposed to have.

Distributing ES6 is just fine, I think that will be the de-facto very soon, but not yet.

**Comment by priyajeet** [comment]
> Huh, that's... inconsistent with my experience. Especially in TypeScript-land, there are a number of popular coding standards that explicitly forbid sub-path imports. Don't think I've ever had a problem with this. We keep regular tabs on it with bundle analysis.

If you lookup, webpack has an open tracking bug with all issues about tree shaking. When react is involved some parts of it breaks from my experience, specially due to stateless components. Since tree shaking cleanup is actually done by uglifyjs and not by the webpack compile itself, things dont get removed because of side effects added by react. I haven't tried the latest webpack and latest uglifyjs, so things may have changed/improved, but in the version I am using if you do not use full paths you will pull in stuff thats unnecessary. Granted that its not a recommended standard, but having unnecessary stuff in there is worse. I tested this and if you look at the following picture (left is using full paths, right is using import de-structuring), notice how the `ContentPicker` in the bundle even though I did not import it anywhere. The code I used to test it is https://github.com/box/box-ui-elements-demo. 

<img width="1864" alt="screen shot 2018-02-20 at 10 53 44 pm" src="https://user-images.githubusercontent.com/1075325/36467087-ff5a307a-1691-11e8-9115-bed5620db675.png">

**Comment by nateabele** [comment]
Interesting. Thanks for the info. I'll look into it.
